### ISSUE TITLE
TriggerFish-Elements

### ISSUE THREAD

**Issue by JTriggerFish** [VCVRack/library#509]
https://github.com/JTriggerFish/TriggerFish-VCV

BSD3 license
VERSION = 0.6.1
Build head commit : bf0bd04315be0cfdb3cca8f578d52d435ef07114

This is a set of 4 modules focusing on analog models, more to come in the future.
Could you please add to the official module list, let me know if there's anything else I need to do or if any issues.
Many thanks!

**Comment by JTriggerFish** [comment]
Please use 45c953e8c4974064cd03257c4c43c787af55384e instead for building as someone found an issue on linux.
Or more generally best to use the head of master for now.

**Comment by cschol** [comment]
This issue needs to be fixed before integration into the plugin manager: https://github.com/JTriggerFish/TriggerFish-VCV/issues/2

**Comment by JTriggerFish** [comment]
Thank you, should be fine now, I fell the code back to SSE2 instead of AVX.
Are there guidelines for what minimum CPUs / instructions sets should be supported ?
For some code SIMD instructions are critical to getting acceptable performance.

**Comment by AndrewBelt** [comment]
The default flags `-march=nocona` should be used.

**Comment by cschol** [comment]
Thank you, works fine now. I will integrate your plugin into the plugin manager. When Andrew builds the next round of updates, it will be picked up and made available. Please use **this issue** to let us know about future updates.

**Comment by cschol** [comment]
@AndrewBelt should he remove the `-msse2` and let it use the default instead? I'll wait with integration until this is done, unless this is OK the way it is.

**Comment by AndrewBelt** [comment]
It doesn't affect anything, just redundant. I would guess that `-std=c++17` will be a problem though.

**Comment by cschol** [comment]
@JTriggerFish Get me the missing information above and I will integrate. Andrew will let you know if there is a problem with the build system.

**Comment by JTriggerFish** [comment]
Ok, thank you !
author: J TriggerFish
authorEmail: FastTriggerFish@gmail.com

Using a pseudonym for work reasons, I presume that's ok.

**Comment by JTriggerFish** [comment]
VERSION = 0.6.2
commit dcd8c9ad9b91f363d785267f32fe86dac2664b1a

Fixed the width of Slop4 that wasn't a multiple of 15 and improved the performance of VDPO by a factor of two.

**Comment by AndrewBelt** [comment]
Updated build

It built, but it's untested.

**Comment by AndrewBelt** [comment]
Updated build

**Comment by JTriggerFish** [comment]
Version 0.6.3
commit 9c9b07663df82ad39b4662e210febde12316688d

-VCA performance improvements
-Better handling of inputs for VDPO to avoid blow ups.
-Higher max bleed for the VCA

Thank you !

**Comment by AndrewBelt** [comment]
Weird, my script didn't copy the build to the package folder, it just threw it away. No idea what happened. Re-updated.

**Comment by JTriggerFish** [comment]
I’m confused about the above, this is an old revision, the one with the commit id I’ve posted above for 0.6.3 ( and 0.6.2 before that ) have the version updated in the makefile.
Is there anything I need to do ?

**Comment by JTriggerFish** [comment]
Understood, thank you.
I've just fixed an issue someone reported for an old version of GCC ( 5.4 ) on linux, to be on the safe side could you please take the latest commit when you update ?
cb4dbe74a73fe48cfbd86a15edf4d162578335b9

Many thanks

**Comment by cschol** [comment]
It has to be fixed with the next round of updates. Sorry for the inconvenience.

**Comment by AndrewBelt** [comment]
Updated build

**Comment by JTriggerFish** [comment]
Version 0.6.4
commit e16db7397695d61dc067f5a31140ecfd201253eb

Fixed the VCA output level that I had accidentally broken in the last version ( output knob had no effect ).
Thank you !

**Comment by AndrewBelt** [comment]
Updated build

**Comment by cschol** [comment]
Updated build.

**Comment by cschol** [comment]
Updated build.

**Comment by JTriggerFish** [comment]
Version 1.0.0
commit 834ad2d7c094d94e51407fa874c6b8c82e26986e

Long overdue update to v1 with the kind contribution of
@chuckhousley

Many thanks

**Comment by JTriggerFish** [comment]
Updated the plugin json to add a donate link, no other changes.
Commit bce959e369cacdccd6f68353166b2d5abc720d50
Thanks!

**Comment by cschol** [comment]
@AndrewBelt Is the donate link used in the app? Should we rev the version for this change?

**Comment by AndrewBelt** [comment]
Yes, the version should be bumped to 1.0.1 or something so there aren't two nonidentical ZIP packages with the same filename floating around.

**Comment by JTriggerFish** [comment]
Understood thanks, done here :
d2a9a2b6ed37fbbe7397f5688a02a62938b71306

**Comment by JTriggerFish** [comment]
Version 2.0
commit e39ef7d886120c77e7471f98a34a12fbe8933973

Thank you !

**Comment by cschol** [comment]
@JTriggerFish please check the SHA provided in comment above. It looks like it points to an old version. Thanks.

**Comment by JTriggerFish** [comment]
@cschol apologies, I copy pasted the wrong line /facepalm
it is :
b454b7388de025ac8c084fe6e49c7633f6b3afe9

**Comment by JTriggerFish** [comment]
Update to version 2.2.0

Commit: 9f3c1804e6d65b6777cf180bbe92cbb9e660f6fc

Tag:
https://github.com/JTriggerFish/TriggerFish-VCV/tree/v2.2.0

Release notes:
https://github.com/JTriggerFish/TriggerFish-VCV/blob/v2.2.0/docs/releases/2.2.0.md

**Comment by cschol** [comment]
Build failure (mac-x64): missing Eigen library headers: https://github.com/JTriggerFish/TriggerFish-VCV/issues/9

**Comment by JTriggerFish** [comment]
Thank you and apologies for this.
The macOS Eigen build failure has been fixed.

Please use the updated commit:

**Commit:** `703ed1ca2b5f324748ee49af2ac699ffb41350e3`  
**Tag:** `v2.2.0`

The vendored Eigen headers were moved outside Rack’s generated `dep` directory, which is removed by `make cleandep`. The complete build now passes natively on both macOS Intel (`x64`) and Apple Silicon (`arm64`), as well as Linux.

CI run: https://github.com/JTriggerFish/TriggerFish-VCV/actions/runs/31954653839

**Comment by JTriggerFish** [comment]
If possible I'd like to supersede the pending request and skip 2.2.0 and release 2.3.0 instead.
Version: 2.3.0
Commit: 42bd76d52382140137e7f49ee2b58f3c53953bc3
Tag: https://github.com/JTriggerFish/TriggerFish-VCV/tree/v2.3.0
Release notes: https://github.com/JTriggerFish/TriggerFish-VCV/blob/v2.3.0/docs/releases/2.3.0.md
CI: https://github.com/JTriggerFish/TriggerFish-VCV/actions/runs/32161449409

**Comment by cschol** [comment]
@JTriggerFish sorry, it just missed the cutoff for the build. It will go out with the next batch.

**Comment by JTriggerFish** [comment]
Hello, I'd like to release 
TriggerFish-Elements 2.5.0

Commit: 472562aa95d2529ef7bcd2fcf4a0707b26165554
Tag: v2.5.0

**Comment by JTriggerFish** [comment]
Last minute sequencer timig fix.

Repository: https://github.com/JTriggerFish/TriggerFish-VCV
Tag: v2.5.1
Commit: f6ce652d9dfb9db6e8af5736edcf3b86aaf1f6a8

**Comment by cschol** [comment]
Static analysis issues: https://github.com/JTriggerFish/TriggerFish-VCV/issues/10

**Comment by JTriggerFish** [comment]
Tag: v2.5.2
Commit: 6a795f76eda7a7a1712a03b145c098bee57225c9
This supersedes 2.5.1 and addresses the static-analysis findings reported in [issue #10](https://github.com/JTriggerFish/TriggerFish-VCV/issues/10).

**Comment by cschol** [comment]
Build failure (mac-x64): too many errors emitted, stopping now [-ferror-limit=]: https://github.com/JTriggerFish/TriggerFish-VCV/issues/11

**Comment by JTriggerFish** [comment]
fixed in TriggerFish Elements 2.5.3.
Commit: 31d9e5feafe1a21fc40bafd167dee590bf1523ac
https://github.com/JTriggerFish/TriggerFish-VCV/commit/31d9e5feafe1a21fc40bafd167dee590bf1523ac
Tag: v2.5.3

**Comment by cschol** [comment]
More compiler issues: re-used this issue and provided a patch: https://github.com/JTriggerFish/TriggerFish-VCV/issues/11#event-30219570799

**Comment by JTriggerFish** [comment]
Thanks for the report and patch. This is fixed in TriggerFish Elements 2.5.4.
Commit: 1a9d99766413dbf74a2d47b16f49f943ef4c8071
Tag: v2.5.4

**Comment by JTriggerFish** [comment]
Hello, could you please update TriggerFish-Elements to 2.5.5.
- Commit: 534cf506153034081b4852b1ce00298ff13aae84
- Tag: [v2.5.5](https://github.com/JTriggerFish/TriggerFish-VCV/tree/v2.5.5)
- Release notes: [2.5.5](https://github.com/JTriggerFish/TriggerFish-VCV/blob/v2.5.5/docs/releases/2.5.5.md)
This update fixes sequencer key handling, chord register drift, slide behavior, and CV interpolation, alongside Electric Piano and Room Reverb performance improvements.
